doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,501 @@
+         :compact:
+
+   .. group-tab:: ``cmake``
+
+      .. zephyr-app-commands::
+         :tool: cmake
+         :app: share/sysbuild
+         :board: reel_board
+         :goals: build
+         :gen-args: -DAPP_DIR=samples/hello_world -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_DEBUG_OPTIMIZATIONS=y -Dmcuboot_DEBUG_OPTIMIZATIONS=y
+         :compact:
+
+See the following subsections for more information.
+
+.. _sysbuild_cmake_namespace:
+
+CMake variable namespacing
+==========================
+
+CMake variable settings can be passed to CMake using ``-D<var>=<value>`` on the
+command line. You can also set Kconfig options via CMake as
+``-DCONFIG_<var>=<value>`` or ``-D<namespace>_CONFIG_<var>=<value>``.
+
+Since sysbuild is the entry point for the build system, and sysbuild is written
+in CMake, all CMake variables are first processed by sysbuild.
+
+Sysbuild creates a namespace for each domain. The namespace prefix is the
+domain's application name. See :ref:`sysbuild_zephyr_application` for more
+information.
+
+To set the variable ``<var>`` in the namespace ``<namespace>``, use this syntax::
+
+  -D<namespace>_<var>=<value>
+
+For example, to set the CMake variable ``FOO`` in the ``my_sample`` application
+build system to the value ``BAR``, run the following commands:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      ::
+
+         west build --sysbuild ... -- -Dmy_sample_FOO=BAR
+
+   .. group-tab:: ``cmake``
+
+      ::
+
+        cmake -Dmy_sample_FOO=BAR ...
+
+.. _sysbuild_kconfig_namespacing:
+
+Kconfig namespacing
+===================
+
+To set the sysbuild Kconfig option ``<var>`` to the value ``<value>``, use this syntax::
+
+  -DSB_CONFIG_<var>=<value>
+
+In the previous example, ``SB_CONFIG`` is the namespace prefix for sysbuild's Kconfig
+options.
+
+To set a Zephyr application's Kconfig option instead, use this syntax::
+
+  -D<namespace>_CONFIG_<var>=<value>
+
+In the previous example, ``<namespace>`` is the application name discussed above in
+:ref:`sysbuild_cmake_namespace`.
+
+For example, to set the Kconfig option ``FOO`` in the ``my_sample`` application
+build system to the value ``BAR``, run the following commands:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      ::
+
+         west build --sysbuild ... -- -Dmy_sample_FOO=BAR
+
+   .. group-tab:: ``cmake``
+
+      ::
+
+        cmake -Dmy_sample_FOO=BAR ...
+
+Kconfig namespaces for the main application
+===========================================
+
+You can pass Kconfig settings to the main Zephyr application in the following ways:
+
+* Formatting Kconfig settings as ``-DCONFIG_<var>=<value>``.
+  This allows you to set Kconfig values at
+  *CMake time* in the same way, whether you are using sysbuild or not.
+* Using the application-specific namespace for the main application.
+
+For example, if ``my_sample`` is the main application, you can control its Kconfig
+values with either ``-DCONFIG_<var>=<value>`` or ``-Dmy_sample_CONFIG_<var>=<value>``.
```

Comment:
Unless I really missed a subtlety in these lines, there is some redundancy with the previous chapter which makes things confusing.

Reply:
good point, let me see if I can adjust the second sentence into a kind of note about `-DCONFIG_<setting>` can also be used for main application.

Reply:
See update here: https://github.com/zephyrproject-rtos/zephyr/compare/ebc0d4ad715399d360d40857fec185763eb1701d..08034b26f59f4d16227e457e8f04f7510f41e8de

Reply:
made it a tip instead of note.

Reply:
Much clearer, thanks